### PR TITLE
Implement rpm-sign

### PR DESCRIPTION
**Pull request by shelbyspees**

### CODE REVIEW

File: lib/revolution.rb [hunk 1]
```diff
@@ -83,6 +105,11 @@ def self.check_pkg(path)
     raise Error::InvalidPackageName, error_msg unless Dir.exist?(path)
   end
 
+  def self.check_rpm(path)
+    basename = File.basename(path)
+    raise Error::InvalidRPM, "#{basename} not found" unless File.exist?(path)
```
Comment (shelbyspees) [inline lib/revolution.rb]: This may not need its own special error class.

File: lib/revolution/rpm_sign.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+require 'open3'
+
+module RPMSign
+  # Expects passphrase file to exist in environment
+  def self.addsign(pkg_file)
+    IO.popen(['bin/sign', pkg_file], &:read)
```
Comment (shelbyspees) [inline lib/revolution/rpm_sign.rb]: The `&:read` part here is equivalent to the following

```ruby
IO.popen(['bin/sign', pkg_file]) do |output|
  output.read
end
```

File: lib/revolution.rb (hunk 1 above)
Comment (shelbyspees) [reply lib/revolution.rb]: I'm keeping it since that's how I've been doing it for every other file type.

File: lib/revolution/rpm_sign.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  end
+
+  def self.signed?(pkg_file)
+    cmd                      = "rpm -qpi #{pkg_file} | grep -Pi '^signature'"
+    stdout, _stderr, _status = Open3.capture3(cmd)
+    false if stdout.include?('(none)')
```
Comment (smaslennikov) [inline lib/revolution/rpm_sign.rb]: As discussed, I'd prefer you search for the Key ID, and fail if it's not there.

**Pull request merged**